packages/trader/src/Modules/Trading/Containers/trade.jsx

```diff
@@ -205,29 +187,60 @@ const ChartMarkers = connect(({ modules, ui, client }) => ({
     currency: client.currency,
 }))(Markers);
 
-class ChartTradeClass extends React.Component {
-    state = {
-        active_category: null,
-    };
-    bottomWidgets = ({ digits, tick }) => <ChartBottomWidgets digits={digits} tick={tick} />;
-    topWidgets = ({ ...props }) => {
-        const { is_digits_widget_active, try_synthetic_indices } = this.props;
+const Chart = props => {
+    const {
+        active_symbols,
+        chart_layout,
+        chart_type,
+        exportLayout,
+        extra_barriers = [],
+        end_epoch,
+        granularity,
+        has_alternative_source,
+        is_trade_enabled,
+        is_socket_opened,
+        main_barrier,
+        refToAddTick,
+        resetRefresh,
+        setChartStatus,
+        settings,
+        show_digits_stats,
+        should_refresh,
+        symbol,
+        wsForget,
+        wsForgetStream,
+        wsSendRequest,
+        wsSubscribe,
+    } = props;
+
+    const charts_ref = React.useRef();
+    const props_ref = React.useRef();
+    props_ref.current = props;
```

Comment:
special thing behind this?

Reply:
Changing reference of topWidgets function by adding dependencies to `useCallback` results in Smartcharts performance drop.
so, using `props_ref` to get the current props value